mmdet/models/anchor_heads/guided_anchor_head.py

```diff
@@ -0,0 +1,493 @@
+        anchor_strides (Iterable): Anchor strides.
+        anchor_base_sizes (Iterable): Anchor base sizes.
+        anchoring_means (Iterable): Mean values of anchoring targets.
+        anchoring_stds (Iterable): Std values of anchoring targets.
+        target_means (Iterable): Mean values of regression targets.
+        target_stds (Iterable): Std values of regression targets.
+        loc_filter_thr (float): threshold to filter out unconcerned regions
+        cls_sigmoid_loss (bool): Whether to use sigmoid loss for
+        classification. (softmax by default)
+        cls_focal_loss (bool): Whether to use focal loss for classification.
+    """  # noqa: W605
+
+    def __init__(self,
+                 num_classes,
+                 in_channels,
+                 feat_channels=256,
+                 octave_base_scale=8,
+                 scales_per_octave=3,
+                 octave_ratios=[0.5, 1.0, 2.0],
+                 anchor_strides=[4, 8, 16, 32, 64],
+                 anchor_base_sizes=None,
+                 anchoring_means=[.0, .0, .0, .0],
+                 anchoring_stds=[1.0, 1.0, 1.0, 1.0],
+                 target_means=(.0, .0, .0, .0),
+                 target_stds=(1.0, 1.0, 1.0, 1.0),
+                 loc_filter_thr=0.01,
+                 loc_focal_loss=True,
+                 cls_sigmoid_loss=False,
+                 cls_focal_loss=False):
+        super(GuidedAnchorHead, self).__init__()
+        self.in_channels = in_channels
+        self.num_classes = num_classes
+        self.feat_channels = feat_channels
+        self.octave_base_scale = octave_base_scale
+        self.scales_per_octave = scales_per_octave
+        self.octave_scales = octave_base_scale * np.array(
+            [2**(i / scales_per_octave) for i in range(scales_per_octave)])
+        self.approxs_per_octave = len(self.octave_scales) * len(octave_ratios)
+        self.octave_ratios = octave_ratios
+        self.anchor_strides = anchor_strides
+        self.anchor_base_sizes = list(
+            anchor_strides) if anchor_base_sizes is None else anchor_base_sizes
+        self.anchoring_means = anchoring_means
+        self.anchoring_stds = anchoring_stds
+        self.target_means = target_means
+        self.target_stds = target_stds
+        self.loc_filter_thr = loc_filter_thr
+        self.loc_focal_loss = loc_focal_loss
+        assert self.loc_focal_loss, 'only focal loss is supported in loc'
+        self.cls_sigmoid_loss = cls_sigmoid_loss
+        self.cls_focal_loss = cls_focal_loss
+
+        self.approx_generators = []
+        self.base_approx_generators = []
+        for anchor_base in self.anchor_base_sizes:
+            self.approx_generators.append(
+                AnchorGenerator(anchor_base, self.octave_scales,
+                                self.octave_ratios))
+            self.base_approx_generators.append(
+                AnchorGenerator(anchor_base, [self.octave_scales[0]], [1.0]))
```

Comment:
We do not need `base_approx_generators` to get the base shapes. meshgrid and \sigma (octave_base_scale) is enough.